common/options/options.go

```diff
@@ -679,7 +679,7 @@ func (opts *ToolOptions) NormalizeOptionsAndURI() error {
 	if err != nil {
 		return err
 	}
-	err = opts.setOptionsFromURI(cs)
+	err = opts.setOptionsFromURI(*cs)
```

Comment:
I think it would make more sense to change this func to just accept the pointer.